### PR TITLE
Fix #11520 - Remove root_path from file_paths.xml and ensure PDF downloads to external SD card work

### PR DESCRIPTION
**Pull request by jumde**
Resolves https://github.com/brave/brave-browser/issues/11520

## Submitter Checklist:

- [x] Submitted a [ticket](https://github.com/brave/brave-browser/issues) for my issue if one did not already exist.
- [x] Used Github [auto-closing keywords](https://help.github.com/articles/closing-issues-via-commit-messages/) in the commit message.
- [ ] Added/updated tests for this change (for new code or code which already has tests).
- Verified that these changes build without errors on
  - [x] Android
  - [ ] iOS
  - [ ] Linux
  - [ ] macOS
  - [ ] Windows
- Verified that these changes pass automated tests (unit, browser, security tests) on
  - [ ] iOS
  - [ ] Linux
  - [ ] macOS
  - [ ] Windows
- [ ] Verified that all lint errors/warnings are resolved (`npm run lint`, `npm run gn_check`)
- [ ] Ran `git rebase master` (if needed).
- [ ] Ran `git rebase -i` to squash commits (if needed).
- [ ] Tagged reviewers and labelled the pull request as needed.
- [ ] Requested a security/privacy review as needed.
- [ ] Added appropriate QA labels (`QA/Yes` or `QA/No`) to the associated issue
- [ ] Added appropriate release note labels (`release-notes/include` or `release-notes/exclude`) to the associated issue
- [ ] Public documentation has been updated as necessary. For instance:
  - [ ] https://github.com/brave/brave-browser/wiki/Deviations-from-Chromium-(features-we-disable-or-remove)
  - [ ] https://github.com/brave/brave-browser/wiki/Proxy-redirected-URLs
  - [ ] https://github.com/brave/brave-browser/wiki/Fingerprinting-Protections
  - [ ] https://github.com/brave/brave-browser/wiki/Brave%E2%80%99s-Use-of-Referral-Codes
  - [ ] https://github.com/brave/brave-browser/wiki/Custom-Headers
  - [ ] https://github.com/brave/brave-browser/wiki/Web-compatibility-issues-with-tracking-protection
  - [ ] https://github.com/brave/brave-browser/wiki/QA-Guide

## Test Plan:
Specified here: https://github.com/brave/security/issues/242

## Reviewer Checklist:

- [ ] New files have MPL-2.0 license header.
- [ ] Request a security/privacy review as needed.
- [ ] Adequate test coverage exists to prevent regressions 
- [ ] Verify test plan is specified in PR before merging to source

## After-merge Checklist:

- [ ] The associated issue milestone is set to the smallest version that the
  changes has landed on.
- [ ] All relevant documentation has been updated.

### CODE REVIEW

File: patches/chrome-browser-download-android-java-src-org-chromium-chrome-browser-download-DownloadManagerBridge.java.patch [hunk 1]
```diff
@@ -0,0 +1,40 @@
+diff --git a/chrome/browser/download/android/java/src/org/chromium/chrome/browser/download/DownloadManagerBridge.java b/chrome/browser/download/android/java/src/org/chromium/chrome/browser/download/DownloadManagerBridge.java
+index 9dee29045bb7d38473529f3dd362e279bb53e692..dc1b84f7ef31e4b65dad9d9b00173a968af7d0cc 100644
+--- a/chrome/browser/download/android/java/src/org/chromium/chrome/browser/download/DownloadManagerBridge.java
++++ b/chrome/browser/download/android/java/src/org/chromium/chrome/browser/download/DownloadManagerBridge.java
```
Comment (jumde) [inline patches/chrome-browser-download-android-java-src-org-chromium-chrome-browser-download-DownloadManagerBridge.java.patch:4]: Will be included in M87: https://chromium.googlesource.com/chromium/src.git/+/6fcac9a226e802cc99e31c025efc41c40f2512c1

Without this patch downloads to external SD card crash.

**Review by SergeyZhukovsky — APPROVED** [review]
++

**Comment by SergeyZhukovsky** [comment]
@samartnik could you please check that when it's merged and on nightly?

**Comment by jumde** [comment]
Created a follow-up issue to track fix for blank pdfs in external storage.

File: patches/chrome-browser-download-android-java-src-org-chromium-chrome-browser-download-DownloadManagerBridge.java.patch [hunk 2]
```diff
@@ -0,0 +1,40 @@
...
++import android.os.Build.VERSION;
++import android.os.Build.VERSION_CODES;
+ import android.os.Environment;
+ import android.text.TextUtils;
+ 
+@@ -90,6 +93,7 @@ public class DownloadManagerBridge {
+             String filePath, long fileSizeBytes, String originalUrl, String referer,
+             String downloadGuid) {
+         assert !ThreadUtils.runningOnUiThread();
++	assert VERSION.SDK_INT < VERSION_CODES.Q : "Deprecated in Q, may cause crash.";
```
Comment (mkarolin) [inline patches/chrome-browser-download-android-java-src-org-chromium-chrome-browser-download-DownloadManagerBridge.java.patch:19]: Chromium patch seems to have 8 leading spaces here.

**Review by mkarolin — APPROVED** [review]
LGTM

**Review by samartnik — APPROVED** [review]
lgtm

**Comment by kjozwiak** [comment]
Verification **PASSED** on `Samsung S10+` running `Android 10` using `Brave 1.18.5 CR: 86.0.4240.111`

Went through the STR/Cases outlined via https://github.com/brave/security/issues/242#issue-722648138 & https://github.com/brave/security/issues/242#issuecomment-714266571.

**Pull request merged**